src/Shell_Command.php

```diff
@@ -25,15 +25,16 @@ class Shell_Command extends EE_Command {
 	 */
 	public function __invoke( $args ) {
 		EE\Utils\delem_log( 'ee shell start' );
-		$args = EE\Utils\set_site_arg( $args, 'shell' );
+		$args      = EE\SiteUtils\auto_site_name( $args, 'shell', '' );
 		$site_name = EE\Utils\remove_trailing_slash( $args[0] );
-		if ( EE::db()::site_in_db( $site_name ) ) {
-			$db_select = EE::db()::select( ['site_path'], ['sitename' => $site_name]);
-			$site_root    = $db_select[0]['site_path'];
-		} else {
+
+		$site = Site::find( $site_name );
```

Comment:
`Site::find` is not a sufficient check. The site needs to be present as well as enabled otherwise the opening it's shell will not be possible. Create a model maybe `Site::enabled` that checks precisely both the things. It is required in almost all the commands.